srcts/src/components/webcomponents/taskButton.ts

```diff
@@ -0,0 +1,82 @@
+import { LitElement, html, css } from "lit";
+import { property } from "lit/decorators.js";
+import { InputBinding, registerBinding } from "../_utils";
+
+export class BslibTaskButtonContents extends LitElement {
+  static tagName = "bslib-task-button-contents";
+  static isShinyInput = false;
```

Comment:
```suggestion
  static isShinyInput = true;
```

FYI, our general setup is to set this to true so that you can keep the input binding methods in the same component class and have `makeInputBinding()` upgrade `BslibTaskButtonContents` with the necessary input binding methods.

I _think_ `makeInputBinding()` covers all of the input binding methods, but there is a chance you're doing something here that isn't supported yet (I haven't read these classes in detail, just sharing a passing thought).

Reply:
It's really unfortunate but it's not possible in this case. The object being bound as an input is _not_ this web component; it's a wrapper `<button>`. I looked at this extensively yesterday and then spoke with @nstrayer extensively about this today, and there are a few problems here:

1. It's a good idea to use a real, native `<button>` element here. If I don't do that, and use a custom element instead, I need to do the work to fully simulate everything about a native button with full fidelity. (For example, I didn't think of the fact that a focused button treats spacebar as a click, which I'd also have to remember to preventdefault)
2. In theory `<button is="task-button">` would let me have a native button that's also a web component. But this doesn't work in WebKit and they have no intentions to ever implement it; and even in browsers that support it, there are restrictions on what you can do with the shadow DOM.
3. We could put a web component as a wrapper around a real `<button>` but you know as well as anyone the trouble that an extra wrapper element can introduce.

Here are the options I sketched out:
https://codepen.io/jcheng5/pen/ZEwNwNb

Options 1 and 3 were the two best options. I would probably do Option 1 if I was starting from scratch now (and Nick agreed). But this is already working and has few downsides besides having two classes instead of one.

Reply:
Oh, okay, that makes sense! I hadn't seen the detail that you were binding the input to a different part of the overall component. That detail might be worth calling out in a comment?

Reply:
Here's where I ended up, should be much clearer now.

```html
<button class="btn btn-primary bslib-task-button" id="go" type="button">
  <bslib-switch-inline case="ready">
    <span slot="ready">Go</span>
    <span slot="busy">
      <i class="fas fa-arrows-rotate fa-spin" role="presentation" aria-label="arrows-rotate icon" aria-hidden="true"></i>
      Processing...
    </span>
  </bslib-switch-inline>
</button>
```

The outermost element is a native `<button>` and that's what is bound, same as an action button basically. The part that switches between states is a generic `<bslib-switch-inline>` custom element (there's also `<bslib-switch>` if you want it to be `display: block`).

The `case` can be any string, and it just renders whatever child has that string as its slot name.